circuit-construction/tests/sudoku.rs

```diff
@@ -55,17 +42,17 @@ where
         };
         let e = -1;
         let data: Vec<Vec<i32>> = vec![
-            vec![5, 3, e, /* */ e, 7, e, /* */ e, e, e],
-            vec![6, e, e, /* */ 1, 9, 5, /* */ e, e, e],
-            vec![e, 9, 8, /* */ e, e, e, /* */ e, 6, e],
+            vec![e, 4, e, /* */ 3, e, e, /* */ e, 8, e],
+            vec![e, e, 3, /* */ 5, e, 2, /* */ e, 4, e],
+            vec![e, e, e, /* */ e, e, 1, /* */ 5, e, 3],
             /* -------------------------------------- */
-            vec![8, e, e, /* */ e, 6, e, /* */ e, e, 3],
-            vec![4, e, e, /* */ 8, e, 3, /* */ e, e, 1],
-            vec![7, e, e, /* */ e, 2, e, /* */ e, e, 6],
+            vec![e, e, 2, /* */ 4, e, e, /* */ 1, e, e],
+            vec![4, e, e, /* */ e, 2, e, /* */ e, e, 7],
+            vec![e, e, 7, /* */ e, e, 3, /* */ 8, e, e],
             /* -------------------------------------- */
-            vec![e, 6, e, /* */ e, e, e, /* */ 2, 8, e],
-            vec![e, e, e, /* */ 4, 1, 9, /* */ e, e, 5],
-            vec![e, e, e, /* */ e, 8, e, /* */ e, 7, 9],
+            vec![1, e, 8, /* */ 2, e, e, /* */ e, e, e],
+            vec![e, 7, e, /* */ 1, e, 8, /* */ 4, e, e],
+            vec![e, 2, e, /* */ e, e, 9, /* */ e, 1, e],
```

Comment:
why are you changing the sudoku :D ? it was not correct? (I took it from wikipedia without really verifying haha)

Reply:
> use a new sudoku example as the previous one seems to have no solution for diagonal rules

ah!